qiskit_experiments/library/randomized_benchmarking/clifford_utils.py

```diff
@@ -375,328 +291,317 @@ def _unpack_num(num, sig):
-                qc, optimization_level=1, basis_gates=self.basis_gates, backend=self._backend
-            )
-            self._transpiled_cliff_layer[1].append(transpiled)
-
-        qc = QuantumCircuit(qr)
-        qc.cx(0, 1)
-        qc.cx(1, 0)
-        qc.cx(0, 1)
-        transpiled = transpile(
-            qc, optimization_level=1, basis_gates=self.basis_gates, backend=self._backend
-        )
-        self._transpiled_cliff_layer[1].append(transpiled)
-
-    def _transpile_cliff_layer_2(self):
-        """Layer 2 consists of a Pauli gate on each qubit {Id, X, Y, Z}.
-        Number of Cliffords == 16."""
-        if self._transpiled_cliff_layer[2] != []:
-            return
-
-        pauli = ("i", XGate(), YGate(), ZGate())
-        for p0, p1 in itertools.product(pauli, pauli):
-            qc = QuantumCircuit(2)
-            if p0 != "i":
-                qc.append(p0, [0])
-            if p1 != "i":
-                qc.append(p1, [1])
-
-            transpiled = transpile(
-                qc, optimization_level=1, basis_gates=self.basis_gates, backend=self._backend
-            )
-            self._transpiled_cliff_layer[2].append(transpiled)
+            raise QiskitError("Wrong param {} for rz in clifford".format(inst.params[0]))
 
-    def create_random_clifford(self, rng: Generator) -> QuantumCircuit:
-        """For 1-qubit, select a random transpiled Clifford.
-
-         Returns:
-            A random Clifford represented as a :class:`QuantumCircuit`.
-
-        Raises:
-            QiskitError: if the number of qubits is greater than 2.
+    return inst.name
+
+
+def _load_clifford_compose_1q():
+    dirname = os.path.dirname(__file__)
+    data = np.load(f"{dirname}/data/clifford_compose_1q.npz")
+    return data["table"]
+
+
+def _load_clifford_inverse_1q():
+    dirname = os.path.dirname(__file__)
+    data = np.load(f"{dirname}/data/clifford_inverse_1q.npz")
+    return data["table"]
+
+
+_CLIFFORD_COMPOSE_1Q = _load_clifford_compose_1q()
+_CLIFFORD_INVERSE_1Q = _load_clifford_inverse_1q()
+
+
+########
+# Functions for 2-qubit integer Clifford operations
+def compose_2q(lhs: Integral, rhs: Integral) -> Integral:
+    """Return the composition of 2-qubit clifford integers."""
+    num = lhs
+    for layer, idx in enumerate(_layer_indices_from_num(rhs)):
+        circ = _CLIFFORD_LAYER[layer][idx]
+        num = _compose_num_with_circuit_2q(num, circ)
+    return num
+
+
+def inverse_2q(num: Integral) -> Integral:
+    """Return the inverse of 2-qubit clifford integers."""
+    return _CLIFFORD_INVERSE_2Q[num]
+
+
+def num_from_2q_circuit(qc: QuantumCircuit) -> Integral:
+    """Convert a given 2-qubit Clifford circuit to the corresponding integer."""
+    return _compose_num_with_circuit_2q(0, qc)
+
+
+# Shortcut to call the function converting circuit to num by number of qubits
+# TODO: Too much? (if it is usefule, add more explanation)
+num_from_circuit = {
+    1: num_from_1q_circuit,
+    2: num_from_2q_circuit,
```

Comment:
Honestly, I prefer more descriptive
```
if num_qubits == 1:
    return num_from_1q_circuit(num)
if num_qubits == 2:
    return num_from_2q_circuit(num)
```
because I can understand what is done at a glance. The shortcut API call below looks awkward to parse for me. 
```
if num_qubits in {1, 2}:
    return num_from_circuit[num_qubits](num)
```
In total, I don't think 2 lines of reduction pays for having one more public API to maintain. Any preference? @nkanazawa1989

Reply:
It's a matter of taste. Do as you prefer.